README.md

```diff
@@ -254,19 +238,115 @@ client := openfeature.NewClient("clientName")
 client.AddHandler(openfeature.ProviderError, &providerErrorCallback)
 ```
 
-### Shutdown: 
+### Shutdown
 
 The OpenFeature API provides a close function to perform a cleanup of all registered providers.
 This should only be called when your application is in the process of shutting down.
 
 ```go
 import "github.com/open-feature/go-sdk/pkg/openfeature"
 
-...
-
 openfeature.Shutdown()
 ```
 
+## Extending
```

Comment:
@thisthat @thomaspoignant @toddbaert @Kavindu-Dodan

Please review these code examples one more time to make sure they're correct.

Reply:
Suggestions 

- Let's mark `Init`, `Status`, `Shutdown` as optional implementations that comes from `StateHandler` interface

- Let's add event handler interface implementation,


```go
// optional(implements EventHandler interface): return the event channel of this provider
func (i MyFeatureProvider) EventChannel() <-chan openfeature.Event {
	// event channel
}
```